param/version.py

```diff
@@ -233,6 +249,8 @@ def __str__(self):
         """
         if self.release is None: return 'None'
         release = '.'.join(str(el) for el in self.release)
+        release = ('%s%sdev%d' % (release, '.' if self.pep440 else '', self.dev)
+                   if self.dev else release)
```

Comment:
Do people do `dev0` (thinking along the lines of 0 being false above)? Actually, I realize now I have seen e.g. `1.x.dev` (numba, pandas). But we can't support everything, and I think just `.dev` doesn't comply with PEP 440 ("The developmental release segment consists of the string .dev, followed by a non-negative integer value").

This is all getting a bit crazy, sorry...